### PR TITLE
Nuttx patches for iotjs automatic build

### PR DESCRIPTION
**Pull request by tadziopazur**
TL;DR: precommit --test=nuttx produces nuttx.bin that works

This change makes the precommit test action on nuttx relevant in that the results can be uploaded into STM32F4 Discovery1 and be run with IoT.js interpreter.

What it does is to patch the nuttx os (works with nuttx-7.19 tag) to fix compilation errors.
During build stm32f4discovery/usbnsh defconfig is overwritten with a working one.

### CODE REVIEW

**Comment by pmarcinkiew** [comment]
LGTM

File: config/nuttx/stm32f4dis/patches/0004-Provide-working-iotjs-config-for-7.19-with.patch [hunk 1]
```diff
@@ -0,0 +1,1818 @@
...
+ -ROMFS, TMPFS  -USB/NSH  -ADC, I2C, SPI, UART, GPIO, TCP/IP, UDP
+
+---
+ configs/stm32f4discovery/usbnshiotjs/Make.defs |  113 ++
+ configs/stm32f4discovery/usbnshiotjs/defconfig | 1589 ++++++++++++++++++++++++
+ configs/stm32f4discovery/usbnshiotjs/setenv.sh |   80 ++
+ 3 files changed, 1782 insertions(+)
+ create mode 100644 configs/stm32f4discovery/usbnshiotjs/Make.defs
+ create mode 100644 configs/stm32f4discovery/usbnshiotjs/defconfig
+ create mode 100755 configs/stm32f4discovery/usbnshiotjs/setenv.sh
```
Comment (hs0225) [inline config/nuttx/stm32f4dis/patches/0004-Provide-working-iotjs-config-for-7.19-with.patch]: How about creating files and copying it without applying a patch file like tizenrt?

Comment (tadziopazur) [inline config/nuttx/stm32f4dis/patches/0004-Provide-working-iotjs-config-for-7.19-with.patch]: I prefer a patch, because the contents, location and validation are kept in a single place.
With file copy, one needs to know:
 - where to copy the file,
 - should there be no file, or overwrite is okay?
And we lose all the validation that patch applying does (I prefer to know the patch does not apply, than just blindly overwrite a file that might have changed).

I would also like to replace our current apps/system/iotjs directory copy with a patch. Maybe later, definitely not in this patch.

Comment (hs0225) [reply config/nuttx/stm32f4dis/patches/0004-Provide-working-iotjs-config-for-7.19-with.patch]: There are too many things in your patch files and it is hard to maintain. If the config file will be changed, how do you manage it? If the version of the nuttx is updated and config file is changed, you will need to delete contents in the patch files and copy text.
The role of your patch will make the file new, not change files. I want to separate it into three files.
And we have copied config file so far. And you should update the documents about the changes.

File: config/nuttx/stm32f4dis/patches/0001-Only-call-up_netinitialize-if-network-is-configured.patch [hunk 2]
```diff
@@ -0,0 +1,25 @@
+From 662e30417dae4bae722a3408deae930a7917d19d Mon Sep 17 00:00:00 2001
+From: Tomasz Wozniak <t.wozniak@samsung.com>
```
Comment (hs0225) [inline config/nuttx/stm32f4dis/patches/0001-Only-call-up_netinitialize-if-network-is-configured.patch]: I don't want your name to remain here.

**Comment by LaszloLango** [comment]
Could you summarize what issues fix this PR in the commit message?

**Comment by LaszloLango** [comment]
It this PR related to #1126?

**Comment by pmarcinkiew** [comment]
This PR was created as response to https://github.com/Samsung/iotjs/issues/1126

File: config/nuttx/stm32f4dis/patches/0004-Provide-working-iotjs-config-for-7.19-with.patch (hunk 1 above)
Comment (tadziopazur) [reply config/nuttx/stm32f4dis/patches/0004-Provide-working-iotjs-config-for-7.19-with.patch]: Moved the config file to config/nuttx/stm32f4dis/dot_usbnshiotjs_defconfig.
Removed nuttx patches 1 (network buildbreak fix) and 4 (usbnshiotjs configuration for stm32f4).
Still two patches are left (missing include in string include chain, missing definition for PWM/Timer configuration). Without these nuttx would not build with the provided config.

File: config/nuttx/stm32f4dis/patches/0001-Only-call-up_netinitialize-if-network-is-configured.patch (hunk 2 above)
Comment (tadziopazur) [reply config/nuttx/stm32f4dis/patches/0001-Only-call-up_netinitialize-if-network-is-configured.patch]: Why?
In case of this file it no longer matters, since I removed the patch.

**Comment by tadziopazur** [comment]
PTAL.

**Comment by LaszloLango** [comment]
@pmarcinkiew in this case the issue shouldn't be closed. Please re-open it and mention the issue number in the commit message. Issues must be closed when they are resolved.

**Comment by yichoi** [comment]
reopen #1126 since I advocate @LaszloLango 's opinion

**Comment by yichoi** [comment]
This is not match current (.) started environment. I am afraid to back to make another test environment.  so please solve #1126 first then think about reopening this.
